CIPs/cip-137.md

```diff
@@ -0,0 +1,200 @@
+    - *rangeNumber* <integer> - splits the stream set space into the given number of ranges
+    - *rangeIndex* <integer> - subscribes only to the given range index
+
+**Returns:**
+
+- *range* <Array<string>> - a Recon message, representing the latest state of the range described by the subscription request. In the form of `[eventid, ahash, eventid]`
+
+---
+
+### `ceramic_unsubscribe`
+
+This method tells the Ceramic node to unsubscribe from a given stream set.
+
+**Params:**
+
+- *ceramicUrl* <string> - the Ceramic URL to unsubscribe from
+
+**Returns:**
+
+- *success* <boolean> - true if the request was successful
+
+---
+
+### `ceramic_listSubscriptions`
+
+List stream subsets that this node is subscribed to.
+
+**Returns:**
+
+* *subscriptions* <array<object>> - a list of all subscriptions
+  * *ceramicUrl* <string> - the Ceramic URL of the subscription
+  * *rangeNumber* <integer> - the range split of the subscription
+  * *rangeIndex* <integer> - the range index of the subscription
+
+---
+
+### `ceramic_poll`
+
+Used by the client to poll for new events. This method uses the Ceramic nodes local ordering of events within the given *streamid*The Ceramic node constructs this ordering based on the order in which it received the events. If you want the global ordering of events you can use the `ceramic_recon` method.
+
+**Params:**
+
+- *ceramicUrl* <string> - the Ceramic URL to poll for
+- *eventOffset* <integer> - the number of events already consumed
+- *duration* <integer> - the amount of time, in seconds, to wait for new data
+
+**Returns:**
+
+- *events* <array<object>> - the list of events
+    - *eventId* <string> - the id of the event
+    - *id* <string> - CID of the InitEvent for this stream
+    - *prev* <array<string>> - CIDs of previous hash linked events
+    - *header* <object> - header for the event
+    - *data* <object> - the data of the event
+    - *timestamp* <integer> - the unixtime this event was timestamped (if it has been)
+- *eventOffset* <integer> - the number of events consumed
+
+---
+
+### `ceramic_recon`
+
+Interact with the Ceramic node using the Recon protocol directly. This allows you to have greater control over the data you consume, but you have to be able to run the Recon algorithm client side.
+
+**Params:**
+
+- *reconRange* <Array<string>> - a Recon message, *`[eventid or ahash]`*
+
+**Returns:**
+
+- *reconRange* <Array<string>> - a Recon message, *`[eventid or ahash]`*
+
+---
+
+### `ceramic_reconPoll`
+
+Same as `ceramic_recon`, but waits for *duration* amount of time in case any new events arrives at the node during this time.
+
+**Params:**
+
+- *reconRange* <Array<string>> - a Recon message, *`[eventid or ahash]`*
+- *duration* <integer> - the amount of time, in seconds, to wait for new data
+
+**Returns:**
+
+- *reconRange* <Array<string>> - a Recon message, *`[eventid or ahash]`*
+
+---
+
+### `ceramic_getRawEvents`
+
+Get the raw IPLD data of a set of events given an array of eventids.
```

Comment:
This is under specified about recursion.
If the event has a data link instead of a data Json should you pull that block also?
How deep should you go?